### PR TITLE
Trigger RESEAL_MIN_TIMER when block is imported

### PR DESCRIPTION
**Pull request by majecty**

### CODE REVIEW

File: test/src/helper/promise.ts [hunk 1]
```diff
@@ -17,3 +17,36 @@
...
+        const result = await p;
+
+        this.waiting[key] -= 1;
+        if (this.waiting[key] === 0) {
+            delete this.waiting[key];
+        }
+        return result;
+    }
+
+    public checkFulfil() {
```
Comment (sgkim126) [inline test/src/helper/promise.ts]: What does this method?
It just prints log and returns nothing?

Comment (majecty) [reply test/src/helper/promise.ts]: @sgkim126 
Hyunsik is right. When running the Tendermint tests, there were many test failures caused by connection error. If we could check the reason for the timeout, we could save a lot of time.

**Comment by jhs7jhs** [comment]
@sgkim126 I'm not sure about triggering the reseal timer in `chain_new_blocks()`. When the transactions of the new block is removed from the mempool? I think we need to trigger the timer when there are some transactions moved from the future pool to the current pool.

**Comment by sgkim126** [comment]
@jhs7jhs AFAIK, they are removed on `chain_new_blocks`. `chain_new_blocks` is called right before calling `Chain::new_blocks()` in `import_sealed_block` and `import_verified_block`.
The future transactions are moved in `remove_old` method, but I'm not sure whether there is no other code path that moves future transactions.

File: test/src/helper/promise.ts (hunk 1 above)
Comment (sgkim126) [reply test/src/helper/promise.ts]: How about making `checkFulfilled` throw an error if there is a waiting promise?

Comment (sgkim126) [reply test/src/helper/promise.ts]: And please use template string instead of string concatenation.

File: test/src/helper/promise.ts [hunk 2]
```diff
@@ -17,3 +17,46 @@
...
+        }
+        this.waiting = {};
+
+        if (timeoutJobs.length > 0) {
+            console.error(
+                `Timeout period is expired while waiting ${timeoutJobs.join(
+                    ", "
+                )}`
+            );
+            throw new Error("");
```
Comment (sgkim126) [inline test/src/helper/promise.ts]: Why do you throw an empty string?
How about throw
```js
`Timeout period is expired while waiting ${timeoutJobs.join(
                    ", "
                )}`
```

Comment (majecty) [reply test/src/helper/promise.ts]: I fixed it.

**Pull request merged**